ludwig/schema/encoders/image_encoders.py

```diff
@@ -284,368 +284,828 @@ class Stacked2DCNNEncoderConfig(BaseEncoderConfig):
+
+@register_encoder_config("squeezenet_torch", IMAGE)
+@dataclass
+class TVSqueezeNetEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["squeezenet_torch"],
+        default="squeezenet_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    intermediate_size: int = schema_utils.PositiveInteger(
-        default=3072,
-        description="Dimensionality of the intermediate (i.e., feed-forward) layer in the Transformer encoder.",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["intermediate_size"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        [
+            "1_0",
+            "1_1",
+        ],
+        default="1_0",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVSqueezeNetEncoder"]["model_variant"],
     )
 
-    initializer_range: float = schema_utils.NonNegativeFloat(
-        default=0.02,
-        description="The standard deviation of the truncated_normal_initializer for initializing all weight matrices.",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["initializer_range"],
+
+@register_encoder_config("swin_transformer_torch", IMAGE)
+@dataclass
+class TVSwinTransformerEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["swin_transformer_torch"],
+        default="swin_transformer_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    layer_norm_eps: float = schema_utils.NonNegativeFloat(
-        default=1e-12,
-        description="The epsilon used by the layer normalization layers.",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["layer_norm_eps"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        [
+            "t",
+            "s",
+            "b",
+        ],
+        default="t",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVSwinTransformerEncoder"]["model_variant"],
     )
 
-    gradient_checkpointing: bool = schema_utils.Boolean(
-        default=False,
-        description="",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["gradient_checkpointing"],
+
+@register_encoder_config("vit_torch", IMAGE)
+@dataclass
+class TVViTEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["vit_torch"],
+        default="vit_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    patch_size: int = schema_utils.PositiveInteger(
-        default=16,
-        description="The image patch size. Each patch is patch_size² pixels. Must evenly divide the image width and "
-        "height.",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["patch_size"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
```

Comment:
Want to double-check if this field is really `Optional`? If so, make sure `allow_none` is `True`.

Reply:
removed Optional